pySNOM/images.py

```diff
@@ -142,29 +145,67 @@ def transform(self, data):
         else:
             return data - norm
 
+class RotatePhase(Transformation):
 
-class Process:
-    def __init__(self, data = None):
-        self.data = data
-        if self.data is not None:
-            self.output = np.zeros(np.shape(self.data))
+    def __init__(self, degree=90.0):
+        self.degree = degree
+
+    def transform(self, amplitudedata, phasedata):
+        # Construct complex dataset
+        complexdata = amplitudedata * np.exp(phasedata*complex(1j))
+        # Rotate and extract phase
+        return np.angle(complexdata*np.exp(np.deg2rad(self.degree)*complex(1j)))
+
+class SelfReference(Transformation):
+
+    def __init__(self, datatype=DataTypes.Phase):
+        self.datatype = datatype
+
+    def transform(self, data, referencedata):
+        if self.datatype == DataTypes.Amplitude:
+            return np.divide(data, referencedata)
+        elif self.datatype == DataTypes.Phase:
+            return data-referencedata
         else:
-            self.output = None
+            # TODO: We should replace this with a datatype error raise
+            print("Self-referencing makes only sense for amplitude or phase data")
```

Comment:
`raise RuntimeError("text")` :) if we are lazy.

Reply:
OK!